var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -241,7 +241,8 @@ class Petsc(Package):
     depends_on('superlu-dist@xsdk-0.2.0+int64', when='@xsdk-0.2.0+superlu-dist+mpi+int64')
     depends_on('superlu-dist@develop~int64', when='@develop+superlu-dist+mpi~int64')
     depends_on('superlu-dist@develop+int64', when='@develop+superlu-dist+mpi+int64')
-    depends_on('mumps+mpi~int64', when='+mumps')
+    depends_on('mumps+mpi~int64~metis~parmetis', when='+mumps~metis')
```

Comment:
I have a doubt on this line: I think it forbids building petsc on top of an already installed mumps with only one of metis and parmetis. For instance instance `spack install petsc+mumps~metis ^mumps+metis~parmetis` or `spack install petsc+mumps~metis ^mumps~metis+parmetis`.
I am not sure if this is OK or not.

Maybe it should remain
```suggestion
    depends_on('mumps+mpi~int64', when='+mumps~metis')
```
What do you think?

Reply:
@balay @jedbrown @BarrySmith

Reply:
Sorry - missed followup on this one.

- `spack install petsc+mumps~metis ^mumps+metis~parmetis` 
-  `spack install petsc+mumps~metis ^mumps~metis+parmetis`

I don't think there is anything in petsc that prevents the above 2 cases  - but probably not worth supporting in spack [due to the extra complexity - shared build might easily work - but static build would need the dependent metis/parmetis libs aswell]

> mumps conflicts with ~mpi, so +mumps implies +mpi

mumps~mpi is supported by petsc - but its not yet in spack perhaps due to the additional complexly of adding this in.

> petsc +metis +mpi depends on metis and parmetis

again this is probably clubbed as a single variant [for metis/parmetis] to avoid dealing with the complexity of dealing with them correctly [with 2 variants]